server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldTypeTests.java

```diff
@@ -492,6 +492,22 @@ public void testRescoreOversampleModifiesNumCandidates() {
         checkRescoreQueryParameters(fieldType, 1000, 1000, 11.0F, OVERSAMPLE_LIMIT, OVERSAMPLE_LIMIT, 1000);
     }
 
+    public void testRescoreOversampleZeroBypassesRescore() {
+        DenseVectorFieldType fieldType = new DenseVectorFieldType(
+            "f",
+            IndexVersion.current(),
+            FLOAT,
+            3,
+            true,
+            VectorSimilarity.COSINE,
+            randomIndexOptionsHnswQuantized(),
+            Collections.emptyMap()
+        );
+
+        Query query = fieldType.createKnnQuery(VectorData.fromFloats(new float[] { 1, 4, 10 }), 10, 100, 0f, null, null, null);
+        assertTrue(query instanceof ESKnnFloatVectorQuery);
```

Comment:
I think we need to check that the query parameters do not include rescoring when 0 is used in the query. Also, we should probably check that we can't set 0 for previous index versions

Reply:
++ I will add a test